### DISCUSSION TITLE
Interface to assembly language?

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1058] [General]
Does this possible? It's possible with gm2: https://www.nongnu.org/gm2/11/assembly_language.html

**Comment by demoitem** [comment]
Its possible. Just use the EXTERNAL pragma as in interfacing to C functions.

But its not recommended writing pure assembly and then calling them from modula3. You
will almost always stuff up the calling convention and risk disaster. Better to write your
functions interfaces in C and then use the asm directive to code up in assembly.
